enumerables.rb

```diff
@@ -0,0 +1,157 @@
+# rubocop:disable Style/CaseEquality, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    i = 0
+    while i < length
+      arr << self[i] if yield(self[i])
+      i += 1
+    end
+    arr
+  end
+
+  def my_all?(aaa = nil)
+    if aaa
+
+      my_each { |elem| return false unless aaa === elem }
+
+    end
+    unless block_given?
+      i = 0
+      while i < length
+        return false unless self[i]
+
+        i += 1
+      end
+      return true
+    end
+    bool = true
+    i = 0
+    while i < length
+      bool = false unless yield(self[i])
+      i += 1
+    end
+    bool
+  end
+end
+
+module Enumerable
+  def my_any?(aaa = nil)
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
true_array = [nil, false, true, []]
true_array.my_any? == true_array.any? #should return true if my_any works as expected
```